File: Frontend/trackr-administrador/src/app/shared/http/seguridad/usuario-locacion.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { UsuarioLocacion } from '@models/seguridad/usuario-locacion';

@Injectable()
export class UsuarioLocacionService {
  private dataUrl = 'usuarioLocacion/'

  constructor(private http: HttpClient) { }

  public consultarPorUsuario(idUsuario: number) {
    return this.http.get<UsuarioLocacion[]>(this.dataUrl + `consultarPorUsuario/${idUsuario}`);
  }

  public consultar(idRol: number) {
    return this.http.get<UsuarioLocacion>(this.dataUrl + `consultar/${idRol}`);
  }

  public agregar(usuarioLocacion: UsuarioLocacion) {
    return this.http.post<void>(this.dataUrl + 'agregar', usuarioLocacion);
  }

  public editar(usuarioLocacion: UsuarioLocacion) {
    return this.http.put<void>(this.dataUrl + 'editar', usuarioLocacion);
  }

  public eliminar(idUsuarioLocacion: number) {
    return this.http.delete<void>(this.dataUrl + `eliminar/${idUsuarioLocacion}`);
  }
}

File: Backend/TrackrAPI/TrackrAPI/Services/Seguridad/UsuarioLocacionService.cs
using TrackrAPI.Dtos.Seguridad;
using TrackrAPI.Models;
using TrackrAPI.Repositorys.Seguridad;
using TrackrAPI.Repositorys.Distribucion;
using TrackrAPI.Repositorys.Catalogo;
using System.Collections.Generic;
using TrackrAPI.Helpers;

namespace TrackrAPI.Services.Seguridad
{
    public class UsuarioLocacionService
    {
        private IUsuarioLocacionRepository usuarioLocacionRepository;
        private UsuarioLocacionValidatorService usuarioLocacionValidatorService;
        private IPerfilRepository perfilRepository;
        private ICompaniaRepository companiaRepository;
        private ITipoEmpleadoRepository tipoEmpleadoRepository;
        private IUsuarioRepository usuarioRepository;
        private IHospitalRepository hospitalRepository;
        public UsuarioLocacionService(IUsuarioLocacionRepository usuarioLocacionRepository,
            UsuarioLocacionValidatorService usuarioLocacionValidatorService,
            IPerfilRepository perfilRepository,
            ICompaniaRepository companiaRepository,
            ITipoEmpleadoRepository tipoEmpleadoRepository,
            IUsuarioRepository usuarioRepository,
            IHospitalRepository hospitalRepository)
        {
            this.usuarioLocacionRepository = usuarioLocacionRepository;
            this.usuarioLocacionValidatorService = usuarioLocacionValidatorService;
            this.perfilRepository = perfilRepository;
            this.companiaRepository = companiaRepository;
            this.tipoEmpleadoRepository = tipoEmpleadoRepository;
            this.usuarioRepository = usuarioRepository;
            this.hospitalRepository = hospitalRepository;
        }

        public IEnumerable<UsuarioLocacionDto> ConsultarPorUsuario(int idUsuario)
        {
            return usuarioLocacionRepository.ConsultarPorUsuario(idUsuario);
        }

        public void Agregar(UsuarioLocacion usuarioLocacion)
        {
            usuarioLocacionValidatorService.ValidarAgregar(usuarioLocacion);
            usuarioLocacionRepository.Agregar(usuarioLocacion);
        }

        public void Editar(UsuarioLocacion usuarioLocacion)
        {
            usuarioLocacionValidatorService.ValidarEditar(usuarioLocacion);
            usuarioLocacionRepository.Editar(usuarioLocacion);
        }

        public void Eliminar(int idUsuarioLocacion)
        {
            var usuarioLocacion = usuarioLocacionRepository.Consultar(idUsuarioLocacion);
            usuarioLocacionRepository.Eliminar(usuarioLocacion);
        }

    }
}